storage/src/robocorp/storage/_requests.py

```diff
@@ -27,17 +27,18 @@ def _needs_retry(exc: BaseException) -> bool:
     # 400 - payload is bad and needs to be changed
     # 401 - missing auth bearer token
     # 403 - auth is in place, but not allowed (insufficient privileges)
+    # 404 - resource does not exist
```

Comment:
Since this is planned to get back into a common place and we might need to replicate the behaviour with other resources as well, not just assets.